src/command_modules/azure-cli-interactive/azclishell/threads.py

```diff
@@ -0,0 +1,45 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import threading
+
+
+class ContinuousPingThread(threading.Thread):
+    """ thread to keep the progress indications running """
+    def __init__(self, func, arg):
+        super(ContinuousPingThread, self).__init__()
+        self.func = func
+        self.arg = arg
+
+    def run(self):
+        import time
+        try:
+            while True:
+                if self.func(self.arg):
+                    time.sleep(4)
+                    break
+                time.sleep(.25)
+        except KeyboardInterrupt:
+            pass
+
+
+class LoadCommandTableThread(threading.Thread):
+    """ a thread that loads the command table """
+    def __init__(self, func, arg):
+        super(LoadCommandTableThread, self).__init__()
+        self.daemon = True
+        self.arg = arg
+        self.func = func
+
+    def run(self):
```

Comment:
Why can't this be implemented simply with a function passed into `Thread` constructor?